### DISCUSSION TITLE
[BUG] - React Hook Form and Server Actions

### DISCUSSION THREAD

**Discussion by ungureanustefan** [vercel/next.js#68121] [Help]
### Summary

I have impletemented React Hook Form with server actions in my Next.js 14 project, but there's 2 issues:

1. 'action' does not trigger the 'handleSubmit' from React Hook Form properly, so I had to make a callback that calls 'handleSubmit' right away and pass it to 'action'. If I don't call 'handleSubmit' right away, it doesn't work, and I still get the error.
2. 'action' does not like callbacks and it throws an error in the DOM. See the screenshots.

<img width="555" alt="image" src="https://github.com/user-attachments/assets/e97827a3-71f3-49f5-8e2b-8bc78a5dae92">

<img width="612" alt="image" src="https://github.com/user-attachments/assets/9be7a142-7757-4ee2-a219-2cbace009cdd">

**Comment by Juneezee** [comment]
I see possibly two ways of doing this

1. Follow React Hook Form way, i.e. `<form onSubmit={onSubmit}>`, 
2. Don't return a callback from your `onSubmit` function

```ts
const onSubmit = (formData: FormData): void => {
  handleSubmit(...)
}
```

**Comment by icyJoseph** [comment]
Use these as reference, they are all links to the same resource, in some way

- https://github.com/ProNextJS/forms-management-yt/blob/main/forms-mgmt-finished/src/app/MailForm.tsx
- https://github.com/orgs/react-hook-form/discussions/11832
- https://www.youtube.com/watch?v=VLk45JBe8L8
